src/main/java/org/kairosdb/core/http/rest/MetricsResource.java

```diff
@@ -505,8 +562,12 @@ public Response corsPreflightDelete(@HeaderParam("Access-Control-Request-Headers
 	@POST
 	@Produces(MediaType.APPLICATION_JSON + "; charset=UTF-8")
 	@Path("/datapoints/delete")
-	public Response delete(String json) throws Exception
+	public Response delete(@Context HttpHeaders httpHeaders, String json) throws Exception
 	{
+
+		Scope scope = activateScope("datapoints_delete", httpHeaders);
```

Comment:
We can try-with-resources on scope activation, removing the need for the finally { } block.